### PR TITLE
[WIP] Addonify

### PR DESCRIPTION
**Pull request by DanielOchoa**
This PR converts ember-localstorage-adapter into a proper ember-cli addon. Here's my process:
 - [x] Terraform the project with an ember-cli generated empty addon.
 - [x] Move over metadata code (package.json info, bower.json info, README.md, etc.)
 - [x] Move over the tests to work on the dummy app. 
 - [x] Move over the addon code.
 - [x] Make tests pass. 
 - [x] Update README.md
 - [x] SUCCESS!

**Note**: I may move over the actual adapter code first before the tests, given that this is sort of murky to do TDD on.

### CODE REVIEW

**Comment by DanielOchoa** [comment]
tests should pass initially since I haven't ported the old tests yet.

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,35 @@
+---
 language: node_js
 node_js:
```
Comment (locks) [inline .travis.yml]: The node.js versions should probably be updated to include more recent ones.
Should the addon support 0.10?

File: addon/adapters/ls-adapter.js [hunk 2]
```diff
@@ -0,0 +1,469 @@
+
```
Comment (locks) [inline addon/adapters/ls-adapter.js]: Extra line.

File: addon/serializers/ls-serializer.js [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+   * @method extractSingle
+   * @private
+   * @param {DS.Store} store the returned store
+   * @param {DS.Model} type the type/model
+   * @param {Array} payload returned JSONs
+   */
+  normalizeArrayResponse: function(store, type, payload) {
+    var response = { data: [], included: [] };
+    var _this = this;
+    payload.forEach(function(json){
```
Comment (locks) [inline addon/serializers/ls-serializer.js]: Arrow function here, to avoid `_this`.

**Comment by DanielOchoa** [comment]
Currently stuck with error:
`TypeError: serializer.normalizeResponse is not a function`.

File: README.md [hunk 4]
```diff
@@ -15,19 +14,21 @@ branch.
...
 
-If you are using Ember Localstorage Adapter within an Ember CLI project you can install it as an addon with the following command:
+export default LSSerializer.extend();
 
-```sh
-ember install:bower ember-localstorage-adapter
+// app/adapters/application.js
+import LSAdapter from 'ember-localstorage-adapter';
+
+export default LSSAdapter.extend({
```
Comment (oscarni) [inline README.md]: You imported as `LSAdapter` not `LSSAdapter` :)

Comment (DanielOchoa) [inline README.md]: Oops.

File: .travis.yml (hunk 1 above)
Comment (kybishop) [inline .travis.yml]: [This](https://github.com/ember-cli/ember-cli/blob/master/.travis.yml) is what is listed in ember-cli's travis.yml, minus the now removed "node" line. I'd imagine you'd want to keep in sync with what ember-cli is doing

Comment (DanielOchoa) [inline .travis.yml]: I used the default as generated by ember-cli (plus I added the extra versions of ember) which I'm not sure that it should mirror the one from ember-cli itself. Maybe it should mirror the one from the generator at master? IDK.

**Pull request merged**